src/Features/Core/Portable/Structure/Syntax/AbstractBlockStructureProvider.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Structure
+{
+    internal abstract class AbstractBlockStructureProvider : BlockStructureProvider
+    {
+        private readonly ImmutableDictionary<Type, ImmutableArray<AbstractSyntaxStructureProvider>> _nodeOutlinerMap;
+        private readonly ImmutableDictionary<int, ImmutableArray<AbstractSyntaxStructureProvider>> _triviaOutlinerMap;
+
+        protected AbstractBlockStructureProvider(
+            ImmutableDictionary<Type, ImmutableArray<AbstractSyntaxStructureProvider>> defaultNodeOutlinerMap,
+            ImmutableDictionary<int, ImmutableArray<AbstractSyntaxStructureProvider>> defaultTriviaOutlinerMap)
+        {
+            _nodeOutlinerMap = defaultNodeOutlinerMap;
+            _triviaOutlinerMap = defaultTriviaOutlinerMap;
+        }
+
+        /// <summary>
+        /// Keep in sync with <see cref="ProvideBlockStructureAsync"/>
+        /// </summary>
+        public override void ProvideBlockStructure(BlockStructureContext context)
+        {
+            try
+            {
+                var syntaxRoot = context.Document.GetSyntaxRootSynchronously(context.CancellationToken);
```

Comment:
No.  This is just the base class for C# and VB.  You won't be using this, and it will stay internal (whereas the main service will become public).